docs/testing-with-codeception.md

```diff
@@ -0,0 +1,746 @@
+- WordPress would not have loaded yet when PHPUnit calls the data provider methods, so don't expect to be able to use any WordPress functions within data provider methods.
+
+[Read more on WPLoader module configuration](https://wpbrowser.wptestkit.dev/modules/wploader#configuration).
+
+#### WPQueries
+
+This module is typically used in integration tests, to make assertions on the database queries made by the global $wpdb object, and it **requires** `WPLoader` module to work. It will set, if not set already, the `SAVEQUERIES` constant to true and will throw an exception if the contstant is already set to a falsy value.
+
+[Read more on WPQueries module configuration](https://wpbrowser.wptestkit.dev/modules/wpqueries#configuration).
+
+#### WPFilesystem
+
+This module is typically used in acceptance and functional tests, it extends the `Filesystem` module adding WordPress-specific configuration parameters and methods. It provides methods to read, write and update the WordPress filesystem directly, without relying on WordPress methods, using WordPress functions or triggering WordPress filters.
+
+One of the handy use cases of this module is scaffolding plugins and themes on the fly in the context of tests and auto-remove them after each test.
+
+[Read more on WPFilesystem module configuration](https://wpbrowser.wptestkit.dev/modules/wpfilesystem#configuration).
+
+#### WPCLI
+
+This module is typically used in acceptance and functional tests to invoke WP-CLI commands, and test their output. It will use its own version of WP-CLI, not the one installed in the machine running the tests!
+
+**Important notes:***
+- By default, wp-browser will only include the `wp-cli/wp-cli` package; this package contains the basic files to run WP-CLI and does not contain all the commands that come with a typical `wp-cli` installation. If you require all the commands that usually come installed with WP-CLI, then you should require the `wp-cli/wp-cli-bundle` package as a development dependency of your project.
+- This module defines the environment variable `WPBROWSER_HOST_REQUEST` to distinguish testing session, Altis will detect that and switch to the test database similar to what happens with acceptance test sessions.
+
+[Read more on WPCLI module configuration](https://wpbrowser.wptestkit.dev/modules/wpcli#configuration).
+
+### Altis helpers
+
+Altis extends Codeception/wp-browser with its own helpers, mainly used to test Altis modules, but can be useful for projects as well, and are available for you to use. Check out the `tests/_helpers` directory within the `dev-tools` package to check out existing helpers and new available functionality.
+
+## Scaffolding
+
+Altis has a command to generate / scaffold tests and related artifacts, through the Codeception `bootstrap` and `generate` subcommands.
+
+### Bootstrapping tests
+
+To bootstrap the tests folder, which will create the five default tests suites:
+
+```sh
+composer dev-tools codecept bootstrap
```

Comment:
Can we make this command use the suite.yml files in this repo as templates?

Reply:
That's the plan, yes.